omAPI/views/test_api.py

```diff
@@ -13,14 +13,26 @@
     'methods': ['GET', 'POST']
 }
 
-@app_views.route('/posts/<int:profile_id>', **settings)
-def posts(profile_id=None):
+@app_views.route('/posts/<int:profile_id>/<int:page>', **settings)
+def posts(profile_id=None, page=1):
     if request.method == 'GET':
         if profile_id is None:
             return ("Profile ID expected")
         info = storage.getPost(profile_id=profile_id)
+        print(info)
+        if len(info) >= page*25+25:
+            print('num1')
+            info = info[page*25:25*page+25]
+        else:
+            ending = {'ending': 'right here m8'}
+            if len(info) >= page*25:
+                info.append(ending)
+                print('num2')
+            else:
+                return json.dumps(ending)
```

Comment:
Above, we are appending a dictionary into a list to return, but here we are returning a dictionary. This creates inconsistency in the responses we are expecting on the other side. There are a few ways to fix this. We should discuss it on a call.